src/Microsoft.AspNetCore.Routing/Matching/Ascii.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.AspNetCore.Routing.Matching
+{
+    internal static class Ascii
+    {
+        // case-sensitive equality comparison when we KNOW that 'a' is in the ASCII range
+        // and we know that the spans are the same length.
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool UnsafeAsciiIgnoreCaseEqual(ReadOnlySpan<char> a, ReadOnlySpan<char> b, int length)
+        {
+            // We're doing unsafe operations here, caller is expected to have already checked
+            // length.
+            ref var charA = ref MemoryMarshal.GetReference(a);
+            ref var charB = ref MemoryMarshal.GetReference(b);
+
+            while (length > 0 && AsciiIgnoreCaseEqual(charA, charB))
```

Comment:
I think it is worth decribing this in a comment. It is simple what it does but a lot of people won't be fimilar with the method of doing it

e.g. interate through characters and do a case insensitive compare until the end of the text